text/0000-re-org-docs.md

```diff
@@ -0,0 +1,265 @@
+│   │   │   │   ├── group.md
+│   │   │   │   └── order.md <- NOTE: missing content!
+│   │   │   ├── extensions
+│   │   │   │   └── extensions.md
+│   │   │   ├── lifecycle
+│   │   │   │   ├── analyze.md
+│   │   │   │   ├── build.md
+│   │   │   │   ├── create.md
+│   │   │   │   ├── detect.md
+│   │   │   │   ├── export.md
+│   │   │   │   ├── launch.md
+│   │   │   │   ├── rebase.md
+│   │   │   │   └── restore.md
+│   │   │   └── metadata
+│   │   └── platform.md
+│   ├── operations
+│   │   ├── build.md
+│   │   ├── rebase.md
+├── for-app-developers
+│   ├── how-to-guides
+│   │   ├── configure-build-env.md
+│   │   ├── declare-source-metadata.md
+│   │   ├── download-sbom.md
+│   │   ├── export-to-oci-layout.md
+│   │   ├── mount-volumes.md
+│   │   ├── specify-buildpacks.md
+│   │   ├── specify-extensions.md
+│   │   ├── specify-processes.md
+│   │   ├── understand-failures.md <- NOTE: missing content!
+│   │   ├── use-cache-image.md
+│   │   ├── use-http-proxy.md
+│   │   ├── use-inline-buildpacks.md
+│   │   └── use-project-descriptor.md
+│   └── tutorials
+│       ├── advanced-build
+│       │   ├── build-for-arm.md
+│       │   ├── build-for-windows.md
+│       │   ├── build-on-podman.md
+│       │   └── reproducibility.md
+│       └── basic-build
+│           ├── basic-build.md
+│           └── build-phases.md
+├── for-buildpack-authors
+│   ├── how-to-guides
+│   │   ├── migrate <- NOTE: moved from the "reference" section
+│   │   │   ├── buildpack-api-0.4-0.5.md
+│   │   │   ├── buildpack-api-0.5-0.6.md
+│   │   │   ├── buildpack-api-0.6-0.7.md
+│   │   │   ├── buildpack-api-0.7-0.8.md
+│   │   │   ├── buildpack-api-0.8-0.9.md
+│   │   │   └── buildpack-api-0.9-0.10.md
+│   │   ├── package-buildpack
+│   │   │   ├── buildpack-toml.md
+│   │   │   ├── package-buildpack.md
+│   │   │   ├── specify-targets.md
+│   │   │   └── with-clear-env.md <- NOTE: missing content!
+│   │   ├── publish-buildpack
+│   │   │   ├── publish-a-buildpack.md
+│   │   │   └── publishing-with-github-actions.md
+│   │   ├── write-buildpack
+│   │   │   ├── add-labels.md <- NOTE: missing content!
+│   │   │   ├── add-sbom.md
+│   │   │   ├── create-layer.md
+│   │   │   ├── defer-plan-entries.md <- NOTE: missing content!
+│   │   │   ├── re-use-layers.md <- NOTE: missing content!
+│   │   │   ├── specify-env
+│   │   │   │   ├── add-to-posix-paths.md <- NOTE: missing content!
+│   │   │   │   ├── for-build.md <- NOTE: missing content!
+│   │   │   │   ├── for-process.md <- NOTE: missing content!
+│   │   │   │   ├── for-run.md <- NOTE: missing content!
+│   │   │   │   └── with-modifier.md <- NOTE: missing content!
+│   │   │   ├── specify-processes.md
+│   │   │   ├── specify-slices.md <- NOTE: missing content!
+│   │   │   └── verify-targets.md
+│   │   └── write-extension
+│   │       └── advanced-extensions.md
+│   └── tutorials <- NOTE: this would be the ruby buildpack where you have to keep updating the script every time
```

Comment:
Is it worth changing this up to use `nodejs` instead of `ruby`?  Also, should we simplify the example?  The current example installs the runtime, installs the package manager, and resolves the packages.  Could we simply install a `nodejs` runtime and build a vanilla application?  This would cover all of our objectives around caching and `exec.d`.

Reply:
Sure! I like it :)

Reply:
Commented in adb31d5